### PR TITLE
Allow extension panel to grow without bounds

### PR DESCRIPTION
**Pull request by mmisol**
### Purpose

Removes the 500 units bound on the extension panel width. The panel can
be enlarged as much as needed, while respecting the minimum size of the
canvas panel, which remains as 200 units.

Also, the extension panel size is remembered during a session. Serves
both as a nice to have and a avoids a splitter bug.

Also, the workspace references extension is fixed so that it adapts to
bigger sizes. It didn't stretch beyond 600 units.

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.
- [x] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [x] This PR modifies some build requirements and the readme is updated

### Reviewers

@QilongTang @reddyashish 

### FYIs

@DynamoDS/dynamo

### CODE REVIEW

**Comment by mmisol** [comment]
Workspace references extension looks like this when stretched:

workspace_references_stretched

**Comment by QilongTang** [comment]
@mmisol Can you install TuneUp and provide a screen shot as well? I suspect changes in the other repo might be needed as well.

**Comment by mmisol** [comment]
@QilongTang TuneUp seems to stretch just fine. Do you know what's up with those failures? They don't seem related to my changes and locally the tests pass just fine...

Screen Shot 2020-09-09 at 4 24 25 PM

**Comment by QilongTang** [comment]
@mmisol I noticed some random failures in @aparajit-pratap 's PR as well, thinking we can ignore those for now

File: src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs [hunk 1]
```diff
@@ -1816,9 +1816,11 @@ private void ExtensionHandle_MouseEnter(object sender, MouseEventArgs e)
 
         private bool libraryCollapsed;
         private bool extensionsCollapsed;
+        private GridLength? extensionsColumnWidth;
 
         // Default side bar width
         private const int defaultSideBarWidth = 200;
+        private const int DefaultExtensionBarWidthMultiplier = 2;
```
Comment (QilongTang) [inline src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs:1823]: can you put some comments on the purpose of this?

**Review by QilongTang — APPROVED** [review]
Looks good to me. The change is a bit simpler than I imagined which is cool

**Pull request merged**